File: src/001.Array/628.MaximumProductOfThreeNumbers/628.三个数的最大乘积.go
/*
 * @lc app=leetcode.cn id=628 lang=golang
 *
 * [628] 三个数的最大乘积
 */

// @lc code=start
func maximumProduct(nums []int) int {
	sort.Ints(nums)
	n := len(nums)

	return max(nums[n-1]*nums[n-2]*nums[n-3], nums[0]*nums[1]*nums[n-1])
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// @lc code=end

File: src/001.Array/628.MaximumProductOfThreeNumbers/628.三个数的最大乘积.js
/*
 * @lc app=leetcode.cn id=628 lang=javascript
 *
 * [628] 三个数的最大乘积
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumProduct = function (nums) {
	const sortNums = nums.sort((a, b) => a - b);
	const n = nums.length;

	return Math.max(
		sortNums[n - 1] * nums[n - 2] * nums[n - 3],
		sortNums[0] * sortNums[1] * sortNums[n - 1]
	);
};
// @lc code=end
